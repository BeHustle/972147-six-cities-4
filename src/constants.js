const CARD_TYPES = [
  `Apartment`,
  `Private room`
];

const Cities = {
  AMSTERDAM: [52.38333, 4.9]
};

const MAP_ZOOM = 12;

const ICON_SIZE = [30, 30];

const ICON_PATH = `img/pin.svg`;

export {CARD_TYPES, Cities, MAP_ZOOM, ICON_SIZE, ICON_PATH};
