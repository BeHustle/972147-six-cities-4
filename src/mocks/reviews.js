export const reviews = [
  {
    id: 1,
    offerId: 1,
    author: {
      name: `Max`,
      avatar: `img/avatar-angelina.jpg`
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima, quis!`,
    rating: 5,
    date: new Date(`2019-12-17`)
  },
  {
    id: 2,
    offerId: 1,
    author: {
      name: `Alex`,
      avatar: `img/avatar-max.jpg`
    },
    text: `Lorem ipsum dolor sit amet!`,
    rating: 3,
    date: new Date(`2020-07-17`)
  },
  {
    id: 3,
    offerId: 3,
    author: {
      name: `William`,
      avatar: ``
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. At ipsa labore libero nam sunt temporibus?`,
    rating: 1,
    date: new Date(`2018-03-25`)
  },
  {
    id: 4,
    offerId: 2,
    author: {
      name: `Lucy`,
      avatar: `img/avatar-angelina.jpg`
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    rating: 4,
    date: new Date(`2020-01-13`)
  },
];
