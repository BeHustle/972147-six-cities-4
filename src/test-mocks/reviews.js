export const reviews = [
  {
    id: 1,
    offerId: 1,
    author: {
      name: `Max`,
      avatar: `img/avatar-angelina.jpg`
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima, quis!`,
    rating: 5,
    date: new Date(`2019-12-17`)
  },
  {
    id: 2,
    offerId: 1,
    author: {
      name: `Alex`,
      avatar: `img/avatar-max.jpg`
    },
    text: `Lorem ipsum dolor sit amet!`,
    rating: 3,
    date: new Date(`2020-07-17`)
  },
  {
    id: 3,
    offerId: 3,
    author: {
      name: `William`,
      avatar: ``
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. At ipsa labore libero nam sunt temporibus?`,
    rating: 1,
    date: new Date(`2018-03-25`)
  },
  {
    id: 4,
    offerId: 2,
    author: {
      name: `Lucy`,
      avatar: `img/avatar-angelina.jpg`
    },
    text: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.`,
    rating: 4,
    date: new Date(`2020-01-13`)
  },
];

export const serverReviews = [
  {
    "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "date": `2019-05-08T14:13:56.569Z`,
    "id": 1,
    "rating": 4,
    "user": {
      "avatar_url": `img/1.png`,
      "id": 4,
      "is_pro": false,
      "name": `Max`
    }
  }
];
