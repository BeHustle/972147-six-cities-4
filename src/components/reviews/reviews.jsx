import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {AuthStatus} from '../../constants.js';
import {Operation as DataOperation} from '../../reducer/data/data.reducer.js';
import {getReviews} from '../../reducer/data/data.selectors.js';
import {getAuthStatus} from '../../reducer/user/user.selectors.js';
import AddReview from '../add-review/add-review.jsx';
import Review from '../review/review.jsx';

class Reviews extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onReviewsMount, offerId} = this.props;
    onReviewsMount(offerId);
  }

  render() {
    const {reviews, authStatus, offerId} = this.props;
    return <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((review) =>
          <Review key={review.id} review={review}/>,
        )}
      </ul>
      {authStatus === AuthStatus.AUTH && <AddReview offerId={offerId} />}
    </section>;
  }
}

Reviews.propTypes = {
  offerId: PropTypes.number.isRequired,
  reviews: PropTypes.arrayOf(PropTypes.exact({
    id: PropTypes.number.isRequired,
    author: PropTypes.exact({
      name: PropTypes.string.isRequired,
      avatar: PropTypes.string,
    }),
    text: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    date: PropTypes.instanceOf(Date),
  })),
  onReviewsMount: PropTypes.func.isRequired,
  authStatus: PropTypes.oneOf(Object.values(AuthStatus)),
};

const mapDispatchToProps = (dispatch) => ({
  onReviewsMount(offerId) {
    dispatch(DataOperation.loadReviews(offerId));
  },
});

const mapStateToProps = (state) => ({
  reviews: getReviews(state),
  authStatus: getAuthStatus(state)
});

export default connect(mapStateToProps, mapDispatchToProps)(Reviews);
