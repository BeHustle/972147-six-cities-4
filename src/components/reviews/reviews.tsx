import * as React from 'react';
import {connect} from 'react-redux';
import {AuthStatus} from '../../constants';
import {Operation as DataOperation} from '../../reducer/data/data.reducer';
import {getReviews} from '../../reducer/data/data.selectors';
import {getAuthStatus} from '../../reducer/user/user.selectors';
import AddReview from '../add-review/add-review';
import Review from '../review/review';
import {ReviewInterface} from "../../types";

interface Props {
  offerId: number;
  onReviewsMount: (offerId: number) => void;
  reviews: Array<ReviewInterface>;
  authStatus: string;
}

class Reviews extends React.PureComponent<Props, {}> {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onReviewsMount, offerId} = this.props;
    onReviewsMount(offerId);
  }

  render() {
    const {reviews, authStatus, offerId} = this.props;
    return <section className="property__reviews reviews">
      <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>
      <ul className="reviews__list">
        {reviews.map((review) =>
          <Review key={review.id} review={review}/>,
        )}
      </ul>
      {authStatus === AuthStatus.AUTH && <AddReview offerId={offerId} />}
    </section>;
  }
}

const mapDispatchToProps = (dispatch) => ({
  onReviewsMount(offerId) {
    dispatch(DataOperation.loadReviews(offerId));
  },
});

const mapStateToProps = (state) => ({
  reviews: getReviews(state),
  authStatus: getAuthStatus(state)
});

export {Reviews};

export default connect(mapStateToProps, mapDispatchToProps)(Reviews);
