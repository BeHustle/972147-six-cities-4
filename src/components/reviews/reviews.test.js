import React from 'react';
import renderer from 'react-test-renderer';
import {AuthStatus} from '../../constants.js';
import {Reviews} from './reviews.jsx';
import {reviews} from '../../test-mocks/reviews.js';

it(`Render Reviews list`, () => {
  const tree = renderer
    .create(<Reviews
      offerId={1}
      reviews={reviews}
      authStatus={AuthStatus.NO_AUTH}
      onReviewsMount={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
