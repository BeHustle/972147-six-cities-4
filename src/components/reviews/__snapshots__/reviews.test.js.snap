// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Reviews list 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      4
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Max"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima, quis!
        </p>
        <time
          className="reviews__time"
          dateTime="2019-12-17"
        >
          December 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Alex"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Alex
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet!
        </p>
        <time
          className="reviews__time"
          dateTime="2020-07-17"
        >
          July 2020
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="William"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar.svg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          William
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. At ipsa labore libero nam sunt temporibus?
        </p>
        <time
          className="reviews__time"
          dateTime="2018-03-25"
        >
          March 2018
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Lucy"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Lucy
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit.
        </p>
        <time
          className="reviews__time"
          dateTime="2020-01-13"
        >
          January 2020
        </time>
      </div>
    </li>
  </ul>
</section>
`;
